docs/content/sips/013-cloud-plugin.md

```diff
@@ -0,0 +1,155 @@
+The intended outcome of this SIP is to provide a set of recommendations to move
+the core logic of `spin login` and `spin deploy` into a separate `spin cloud`
+plugin, while continuing to provide a great out-of-the-box experience for those
+wanting to deploy their applications to the Fermyon Cloud.
+
+## Proposal
+
+This document proposes to move two commands from the Spin command-line interface
+to a separate plugin:
+
+1. `spin login`
+2. `spin deploy`
+
+These commands will be moved to a separate `spin cloud` plugin. This document
+will also recommend possible approaches to ensure these commands remain easily
+accessible to new and existing users of the Spin command-line interface.
+
+This enables the Spin team to release the core Spin runtime and the Spin
+command-line interface as stable, while also enabling the functionality of the
+Fermyon Cloud to change and iterate over time.
+
+## Rationale
+
+Several commands were introduced to the Spin CLI, making it very simple for
+someone to deploy their application to the Fermyon Cloud:
+
+- `spin login`
+- `spin deploy`
+
+These commands are orthogonal to the concerns of Spin's core runtime. These
+commands involve the packaging and distribution of a Spin application to the
+Fermyon Cloud. They are considered "adjacent" to the Spin user experience as
+they do not assist the developer with writing their application, nor do they
+relate to Spin's core runtime; they provide a simple experience to ship their
+application to the Fermyon Cloud.
+
+`spin login` and `spin deploy` communicate with the Fermyon Cloud API. As new
+features are introduced to the Fermyon Cloud, the API may change and evolve over
+time.
+
+Building a simple, delightful on-ramp experience for the Fermyon Cloud remains
+top priority. It is especially important that we continue to provide a very
+simple on-ramp experience so users can readily deploy their Spin applications to
+a production-grade system.
+
+Spin’s existing plug-in system allows the Spin community to add and extend the
+functionality of Spin without introducing changes to the Spin command-line
+interface. Plug-ins are described in more detail in [SIP 006 - Spin
+Plugins](./006-spin-plugins.md). This allows us to ship `spin login` and `spin
+deploy` as separate, discrete functionality while ensuring users can still
+access its functionality through familiar tooling.
+
+## Specification
+
+The proposal herein suggests that we re-release the core logic of `spin login`
+and `spin deploy` under a separate `spin cloud` plugin. Spin will alias the
+existing `spin login` and `spin deploy` commands to their respective `spin
+cloud` counterparts to retain backwards compatibility.
+
+When a user executes `spin login` or `spin deploy` and the cloud plugin is not
+installed on their system, Spin will inform the user that the `spin cloud`
+plugin has not been installed.
+
+```
+$ spin login
+Error: plugin `cloud` is not installed. Try running `spin plugin install cloud`.
+```
+
```

Comment:
@bacongobbler should we add a "Distribution" section that lays out the following strategies for distributing the plugin with Spin? We could also merge the update FAQ into this too. Here is a quick take at what it could include
```suggestion
## Distribution
Rather than prompting users to install the plugin on first use of `cloud` commands, ideally the cloud plugin would be delivered with the Spin CLI package. Users should be able to get Spin through one of the following avenues:
1. The install script
1. Building from source
1. Package manager (coming soon)

The install script will be updated to automatically install the cloud plugin and packagings of Spin for package managers will be encouraged to do the same. However, in the case of local builds, upon use of the cloud plugin, users will be prompted of the need to install the cloud plugin and it will be subsequently automatically installed.

The plugin will be reinstalled to the latest compatible version with each new use of the install script or update of the package through a package manager. The plugin will also be able to be updated outside of a Spin update through `spin plugin update && spin plugin upgrade`
```

Reply:
Definitely out of scope for this PR (and perhaps something to consider at the plugin level): do we want to explore an update check mechanism that would prompt a user to consider installing a new version (and help the user install it)?

Reply:
(missed the section below on updates)

Reply:
issue to track for this https://github.com/fermyon/spin/issues/1543